pkg/cloudprovider/provider/aws/provider.go

```diff
@@ -113,24 +113,24 @@ var (
 )
 
 type RawConfig struct {
-	AccessKeyID     providerconfig.ConfigVarString `json:"accessKeyId"`
-	SecretAccessKey providerconfig.ConfigVarString `json:"secretAccessKey"`
+	AccessKeyID     providerconfig.ConfigVarString `json:"accessKeyId,omitempty"`
+	SecretAccessKey providerconfig.ConfigVarString `json:"secretAccessKey,omitempty"`
 
-	Region           providerconfig.ConfigVarString `json:"region"`
-	AvailabilityZone providerconfig.ConfigVarString `json:"availabilityZone"`
+	Region           providerconfig.ConfigVarString `json:"region,omitempty"`
```

Comment:
According to https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required only optional fields should use `omitempty`.

Reply:
But none of these fields is required as they can be populated via the env

Reply:
Only a subset of the fields are being fetched from the env.
Region is an example

Reply:
But it could still come from a configmap or secret or not?

Reply:
Which means the field is not "empty"